cmd/collect-profiles/main.go

```diff
@@ -0,0 +1,364 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/operator-framework-olm/pkg/profiling/config"
+)
+
+const (
+	profileConfigMapLabelKey = "olm.openshift.io/pprof"
+)
+
+var (
+	rootCmd = newCmd()
+
+	// Used for flags
+	namespace       string
+	configMountPath string
+	certMountPath   string
+)
+
+func init() {
+	rootCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "default", "The Kubernetes namespace where the generated configMaps should exist. Defaults to \"default\".")
+	rootCmd.MarkFlagRequired("namespace")
+	rootCmd.PersistentFlags().StringVarP(&configMountPath, "config-mount-path", "c", "/etc/config", "The path to the collect-profiles configuration file.")
+	rootCmd.MarkFlagRequired("config-mount-path")
+	rootCmd.PersistentFlags().StringVarP(&certMountPath, "cert-mount-path", "", "/var/run/secrets/serving-cert", "The path to the tls cert used by the client making https requests against the pprof URLs.")
+}
+
+func main() {
+	Execute()
+}
+
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		klog.Fatal(err)
+		os.Exit(1)
+	}
+}
+
+func getTruePointer() *bool {
+	trueBool := true
+	return &trueBool
+}
+
+func newCmd() *cobra.Command {
+	var cfg config.Configuration
+	return &cobra.Command{
+		Use:   "collect-profiles configMapName:url",
+		Short: "Retrieves the pprof data from a UNL and stores it in a configMap.",
+		Long: `The collect-profiles command makes https requests against pprof URLs
+		provided as arguments and stores that information in immutable configMaps.`,
+		Version: "0.0.1",
+		PersistentPreRunE: func(*cobra.Command, []string) error {
+			return cfg.Load()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if len(args) == 0 {
+				klog.Info("No arguments provided, exiting")
+				return nil
+			}
+
+			jobConfig, err := config.GetConfig(configMountPath)
```

Comment:
What happens when the configMountPath variable (or even the default /etc/config location) doesn't exist? It seems like it would produce an error attempting to open that file when reading the code in the config package. Is that the expected behavior we'd like, or would we rather handle that error and attempt to create that path for them?

Reply:
So when shipped with openshift and using the manifests generated by `make manifests`, CVO will always ensure that the default jobConfig configMap exists. If a user chose to delete this map, it would normalize with a subsequent run.

Reply:
It seems reasonable to return a default jobConfig if the file isn't found.

Reply:
On second thought, it seems better for the job to exit if a user provides an invalid configuration. 

Consider the only configuration available to users, `disabled = true`. When this field is set to true, the container "completes" and exits gracefully. If a user "disabled" the job incorrectly (ie, `false` instead of `False`), by defaulting to values the container would still exit gracefully and the user may not know that the job was not disabled. Alternatively, the existing implementation raises an container Error.